src/App.tsx

```diff
@@ -1,26 +1,393 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useEffect,
+  useRef,
+  useState,
+  Dispatch,
+  SetStateAction,
+} from 'react';
+import { USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { client } from './utils/fetchClient';
+import cN from 'classnames';
+import { TodoItem } from './components/TodoItem';
+import { Filter } from './types/Filter';
+import { Error } from './types/Error';
+import { ErrorNotification } from './components/ErrorNotification';
 
-const USER_ID = 0;
+function filterTodosByStatus(todos: Todo[], completedStatus: boolean | null) {
+  if (completedStatus === null) {
+    return todos;
+  }
+
+  return todos.filter(todo => todo.completed === completedStatus);
+}
+
+function removeTodoById(todos: Todo[], id: number) {
+  return todos.filter(todo => todo.id !== id);
+}
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<Error | null>(null);
+  const [activeFilter, setActiveFilter] = useState<Filter>(Filter.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [todoToDeleteIds, setTodoToDeleteIds] = useState<number[] | null>(null);
```

Comment:
do u need `null` in this case?

```suggestion
  const [todoToDeleteIds, setTodoToDeleteIds] = useState<number[]>([]);
```